src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.SingleDiagnosticKindPullTaggerProvider.cs

```diff
@@ -113,21 +110,6 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
 
             try
             {
-                // If this is not the tagger for compiler-syntax, then suppress diagnostics until the project has fully
-                // loaded.  This prevents the user from seeing spurious diagnostics while the project is in the process
-                // of loading.  We do keep compiler-syntax as that's based purely on the parse tree, and doesn't need
-                // correct project info to get reasonable results.
-                if (_diagnosticKind != DiagnosticKind.CompilerSyntax)
-                {
-                    var service = project.Solution.Services.GetRequiredService<IWorkspaceStatusService>();
-                    if (!await service.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false))
-                        return;
-
-                    using var _ = PooledHashSet<Project>.GetInstance(out var seenProjects);
-                    if (!HasSuccessfullyLoaded(document.Project, seenProjects))
```

Comment:
We do not need this check in the tagger as this is already checked in the core executor: https://github.com/dotnet/roslyn/blob/42bd85d0797efa7fc59a7f0285e35b7501a1e014/src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs#L129-L141